### ISSUE TITLE
Swift 4 dictionary default failure

### ISSUE THREAD

**Issue by yonaskolb** [nicklockwood/SwiftFormat#185]
I'm getting an error when using the new default syntax for dictionaries in Swift 4

```
warning: unexpected token default at ...
error: failed to format any files
```

The offending line of code:
```
carthageFrameworksByPlatform[target.platform.carthageDirectoryName, default: []].insert(fileReference)
```

**Comment by nicklockwood** [comment]
That line on its own seems to work fine for me. Can you paste in the whole file? Or enough of the surrounding context to repro anyway.

**Comment by yonaskolb** [comment]
Seems to be if the line is in a switch statement

```
switch a {
case .a:
    dictionary["a", default: []].insert("b")
}
```

**Comment by nicklockwood** [comment]
Fixed in 0.30.1, thanks!

**Comment by yonaskolb** [comment]
🎉
